### PR TITLE
Update handling of Project EvaluatorSettings for Pkl 0.26.0

### PR DESCRIPTION
**Pull request by HT154**
Currently, using pkl-go to evaluate using a PklProject always yields a warning:
```
warn: Cannot find field on Go struct `pkl.ProjectEvaluatorSettings` matching Pkl property `http`. Ensure the Go structs are up to date with Pkl classes either through codegen or manually adding `pkl` tags.
```

The `http` field is populated to avoid the above warning.

### CODE REVIEW

File: pkl/project.go [hunk 1]
```diff
@@ -46,8 +46,8 @@ type ProjectPackage struct {
 	Uri                 string   `pkl:"uri"`
 }
 
-// ProjectEvaluatorSettings is the Go representation of pkl.Project#EvaluatorSettings
-type ProjectEvaluatorSettings struct {
+// EvaluatorSettings is the Go representation of pkl.EvaluatorSettings
+type EvaluatorSettings struct {
```
Comment (bioball) [inline pkl/project.go:50]: I don't feel great about the name clash between `EvaluatorOptions` and `EvaluatorSettings`.

Some other ideas:

* Move all of these to a sub-package; e.g. `stdlib` or something that.
* Name it `PklEvaluatorSettings`

I think let's keep this name for now, and we can handle this in another PR?

File: pkl/project_test.go [hunk 2]
```diff
@@ -108,7 +119,7 @@ func TestLoadProject(t *testing.T) {
 
 		t.Run("evaluatorSettings", func(t *testing.T) {
 			fals := false
-			expectedSettings := &pkl.ProjectEvaluatorSettings{
+			expectedSettings := &pkl.EvaluatorSettings{
```
Comment (bioball) [inline pkl/project_test.go:122]: Two issues here (per failing tests):

1. The evaluator settings should have `Http`
2. Rewrite test to pass on Pkl 0.25 (probably need different tests for the different versions)

File: pkl/project.go [hunk 3]
```diff
@@ -57,6 +57,7 @@ type ProjectEvaluatorSettings struct {
 	Timeout            Duration          `pkl:"timeout"`
 	ModuleCacheDir     string            `pkl:"moduleCacheDir"`
 	RootDir            string            `pkl:"rootDir"`
+	Http               *Http             `pkl:"http"`
```
Comment (bioball) [inline pkl/project.go:60]: Let's use a different struct, because these are slightly different (CA certificate options doesn't exist here).

**Review by bioball — CHANGES_REQUESTED** [review]

File: pkl/project_test.go [hunk 4]
```diff
@@ -171,3 +196,38 @@ func TestLoadProject(t *testing.T) {
...
 	}
 }
+
+func TestLoadProjectWithProxy(t *testing.T) {
+	manager := NewEvaluatorManager()
+	version, err := manager.(*evaluatorManager).getVersion()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if pklVersion0_26.isGreaterThan(version) {
```
Comment (HT154) [inline pkl/project_test.go:206]: I had to hoist this file into the `pkl` package to access this private member, hence most of the diff in the existing test.

**Review by bioball — APPROVED** [review]
LGTM, thank you!

**Pull request merged**